File: src/main/java/com/example/messmanagement/model/Order.java
package com.example.messmanagement.model;

import jakarta.persistence.*;

@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false)
    private String items; // e.g., "Idli, Vada"

    @Column(nullable = false)
    private Integer quantity;

    @Column(nullable = false)
    private Boolean paymentStatus = false; // Paid or not

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }
    public String getItems() { return items; }
    public void setItems(String items) { this.items = items; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public Boolean getPaymentStatus() { return paymentStatus; }
    public void setPaymentStatus(Boolean paymentStatus) { this.paymentStatus = paymentStatus; }
}

File: app/src/main/java/com/example/messmanagement/model/Order.kt
package com.example.messmanagement.model

data class Order(
    val id: Long? = null, // Optional, set by the server
    val user: User? = null, // Optional, set by the server
    val items: String, // Comma-separated string of items (e.g., "Item1 x2, Item2 x1")
    val quantity: Int, // Total quantity of items
    val paymentStatus: Boolean // Payment status (true/false)
)